src/views/postCreation/PostAuthorSelection/PostAuthorSelection.spec.tsx

```diff
@@ -0,0 +1,172 @@
+import userEvent from '@testing-library/user-event';
+import { PostAuthorSelection } from './PostAuthorSelection';
+
+const onAuthorTableClick = () => {};
+const authors = [
+  {
+    avatar: 'abc',
+    firstName: 'Max',
+    id: 1,
+    lastAddedPost: {
+      id: 11,
+      title: 'about',
+    },
+    lastName: 'Petrenko',
+    mainDirection: {
+      id: 11,
+      name: 'flue',
+    },
+    mainInstitution: {
+      city: {
+        id: 11,
+        name: 'Kyiv',
+      },
+      id: 11,
+      name: 'Bolnitsa #4',
+    },
+    qualification: 'Therapist',
+  },
+  {
+    avatar: 'abc',
+    firstName: 'Tax',
+    id: 2,
+    lastAddedPost: {
+      id: 11,
+      title: 'about',
+    },
+    lastName: 'Ivanov',
+    mainDirection: {
+      id: 11,
+      name: 'flue',
+    },
+    mainInstitution: {
+      city: {
+        id: 11,
+        name: 'Kyiv',
+      },
+      id: 11,
+      name: 'Bolnitsa #4',
+    },
+    qualification: 'Therapist',
+  },
+  {
+    avatar: 'abc',
+    firstName: 'Pax',
+    id: 3,
+    lastAddedPost: {
+      id: 11,
+      title: 'about',
+    },
+    lastName: 'Avdeev',
+    mainDirection: {
+      id: 11,
+      name: 'flue',
+    },
+    mainInstitution: {
+      city: {
+        id: 11,
+        name: 'Kyiv',
+      },
+      id: 11,
+      name: 'Bolnitsa #4',
+    },
+    qualification: 'Therapist',
+  },
+];
+
+describe('PostAuthorSeclection', () => {
+  afterEach(cleanup);
+
+  it('show authors lastnames recieved from props', () => {
+    const { getAllByTestId } = render(
+      <PostAuthorSelection
+        authors={authors}
+        onAuthorTableClick={onAuthorTableClick}
+        handleOnChange={() => {}}
+      />,
+    );
+    const authorNames = getAllByTestId('item').map((tc) => tc.textContent);
+    const fakeAuthorNames = authors.map((author) => author.lastName);
+    expect(authorNames).toEqual(fakeAuthorNames);
+  });
+
+  it('render table once author existed', async () => {
+    const handleOnChangeMock = jest.fn();
+    render(
+      <PostAuthorSelection
+        authors={authors}
+        searchValue="Ivan"
+        onAuthorTableClick={onAuthorTableClick}
+        handleOnChange={handleOnChangeMock}
+      />,
+    );
+    screen.debug();
+    expect(screen.queryByText(/first name/i));
+    expect(await screen.findByText(/first name/i));
+  });
+
+  describe('When the input value is changed', () => {
+    it('handle input changes having an appropriate value', () => {
+      const handleOnChangeMock = jest.fn();
+      const { getByPlaceholderText } = render(
+        <PostAuthorSelection
+          authors={authors}
+          searchValue="Ivan"
+          onAuthorTableClick={onAuthorTableClick}
+          handleOnChange={handleOnChangeMock}
+        />,
+      );
+      screen.debug();
+      const input = getByPlaceholderText(
+        'Choose some author',
+      ) as HTMLInputElement;
+      fireEvent.change(input);
```

Comment:
suggestion: I think it's advised to use `userEvents`

https://testing-library.com/docs/ecosystem-user-event